### DISCUSSION TITLE
TrailingStrategy creating multiple trades per bar

### DISCUSSION THREAD

**Discussion by jpcoder2** [kernc/backtesting.py#272] [Q&A]
When I inherit from TrailingStrategy I am seeing multiple trades per bar created.  Is there something simple I am doing wrong?  Is there an example of this base class that shows it's correct usage?  I've seen the one that uses TrailingStrategy and SignalStrategy together but I just want to use TrailingStrategy.

**Comment by jpcoder2** [comment]
I'm still not sure why this was doing this, but I seemed to have worked around the problem.  I change the condition on my buys and sells from checking self.position.is_long/short to self.position.size == 0.  I only wanted one trade at a time anyway so this seems better, but I still don't get how it could have created multiple trades in the same direction previously.

**Comment by kernc** [comment]
> I am seeing multiple trades per bar created

Just to confirm, were you maybe initializing it with _lots_ of data? Plotting does some downsampling if more than 10_000 candles:
https://github.com/kernc/backtesting.py/blob/73e1534428fc2db9e681030862133b5067508520/backtesting/_plotting.py#L89

Multiple trades per bar have been confirmed with [trade data](https://kernc.github.io/backtesting.py/doc/examples/Quick%20Start%20User%20Guide.html#Trade-data)?
